From dfa8d7d3cff6475056e278952c9e35b2d67ce8c6 Mon Sep 17 00:00:00 2001
From: Ryo Nakano <ryonakaknock3@gmail.com>
Date: Wed, 14 Aug 2024 07:09:21 +0900
Subject: [PATCH] meson: Disable styles

Granite bundles stylesheets for some widgets and that requires
addional dependencies, sassc. However, these stylesheets
don't seem to be applied unless the host system uses elementary
stylesheet.

We don't expect elementary OS users install the app from Flathub,
so disable building stylesheet to lessen number of dependencies
and package size.
---
 lib/Init.vala   | 4 ++++
 lib/meson.build | 2 --
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/lib/Init.vala b/lib/Init.vala
index 98472c58..1c1253a3 100644
--- a/lib/Init.vala
+++ b/lib/Init.vala
@@ -5,8 +5,10 @@
 
 namespace Granite {
     private static bool initialized = false;
+#if 0
     private static Gtk.CssProvider? base_provider = null;
     private static Gtk.CssProvider? dark_provider = null;
+#endif
     private static Gtk.CssProvider? app_provider = null;
 
     /**
@@ -62,6 +64,7 @@ namespace Granite {
             }
         }
 
+#if 0
         if (prefer_dark_style) {
             if (base_provider != null) {
                 Gtk.StyleContext.remove_provider_for_display (display, base_provider);
@@ -85,6 +88,7 @@ namespace Granite {
 
             Gtk.StyleContext.add_provider_for_display (display, base_provider, Gtk.STYLE_PROVIDER_PRIORITY_THEME);
         }
+#endif
     }
 
     private static Gtk.CssProvider? init_provider_from_file (File file) {
diff --git a/lib/meson.build b/lib/meson.build
index 25d4a133..2fa37ce0 100644
--- a/lib/meson.build
+++ b/lib/meson.build
@@ -1,5 +1,4 @@
 subdir('Icons')
-subdir('Styles')
 
 libgranite_sources = files(
     'DateTime.vala',
@@ -63,7 +62,6 @@ libgranite = library(
 
     libgranite_sources,
     icons_resource,
-    stylesheet_resource,
     config_vala,
 
     dependencies: [
-- 
2.43.0

